oracle-plugin/src/e2e-test/java/io.cdap.plugin/oracle/stepsdesign/Oracle.java

```diff
@@ -52,4 +57,35 @@ public void validateTheValuesOfRecordsTransferredToTargetTableIsEqualToTheValues
                          "of the records in the source table", recordsMatched);
   }
 
+  @Then("Enter Oracle plugin with connection arguments {string}")
+  public void enterOraclePluginWithConnectionArguments(String jsonConnectionArgumentsField) {
+    OracleActions.enterConnectionArguments(jsonConnectionArgumentsField);
+  }
+
+  @Then("Verify Oracle plugin in-line error message for incorrect Reference Name: {string}")
+  public void verifyOraclePluginInLineErrorMessageForIncorrectReferenceName(String fields) {
+    OracleActions.verifyGroupByPluginPropertyInlineErrorMessageForRow
+      ("referenceName",
+       PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_INVALID_REFERENCE_NAME)
+         .replace("REFERENCE", PluginPropertyUtils.pluginProp(fields)));
+    OracleActions.verifyGroupByPluginPropertyInlineErrorMessageForColor("referenceName");
+  }
+
+  @Then("Validate OUT record count is equal to records transferred to target BigQuery table")
+  public void validateOUTRecordCountIsEqualToRecordsTransferredToTargetBigQueryTable()
+    throws IOException, InterruptedException, IOException {
+    int targetBQRecordsCount = BigQueryClient.countBqQuery(PluginPropertyUtils.pluginProp("bqTargetTable"));
+    BeforeActions.scenario.write("No of Records Transferred to BigQuery:" + targetBQRecordsCount);
+    Assert.assertEquals("Out records should match with target BigQuery table records count",
+                        CdfPipelineRunAction.getCountDisplayedOnSourcePluginAsRecordsOut(), targetBQRecordsCount);
+  }
+
+  @Then("Verify Oracle plugin in-line error message for incorrect Table Name: {string}")
+  public void verifyOraclePluginInLineErrorMessageForIncorrectTableName(String fields) {
+    OracleActions.verifyGroupByPluginPropertyInlineErrorMessageForRow
+      ("tableName",
+       PluginPropertyUtils.errorProp(E2ETestConstants.ERROR_MSG_INVALID_TABLE_NAME)
+         .replace("TABLE", PluginPropertyUtils.pluginProp(fields)));
+    OracleActions.verifyGroupByPluginPropertyInlineErrorMessageForColor("tableName");
+  }
```

Comment:
Why are we writing these steps in here ? All the above steps are present in cdap-e2e-test repo. Reuse the steps from the framework and remove all the unnecessary Action,Steps and Locators.

Reply:
Reused steps from framework.